person.rb

```diff
@@ -0,0 +1,23 @@
+class Person
+  attr_reader :id
+  attr_accessor :name, :age
+
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  private
+
+  def of_age?
+    @age >= 18
+  end
+
+  public
+
+  def can_use_services?
+    of_age? || @parent_permission
+  end
```

Comment:
[Optional] I suggest that you define the public methods after the constructor since the methods are public by default and then define private methods afterwards. That will make your code more concise.

Reply:
Okay thanks!